rasa/core/validator.py

```diff
@@ -57,13 +58,40 @@ def verify_intents(self, ignore_warnings: bool = True) -> bool:
 
         return everything_is_alright
 
+    def verify_example_repetition_in_intents(
+        self, ignore_warnings: bool = True
+    ) -> bool:
+        """Checks if there is no duplicated example in different intents."""
+
+        everything_is_alright = True
+
+        duplication_hash = defaultdict(set)
+        for example in self.intents.intent_examples:
+            text = example.text
+            duplication_hash[text].add(example.get("intent"))
+
+        for text, intents in duplication_hash.items():
+
+            if len(duplication_hash[text]) > 1:
+                everything_is_alright = ignore_warnings and everything_is_alright
+                logger.warning(
+                    "The example '{}' was found in these multiples intents: {}".format(
+                        text, ", ".join(sorted(intents))
+                    )
+                )
+        return everything_is_alright
+
     def verify_intents_in_stories(self, ignore_warnings: bool = True) -> bool:
         """Checks intents used in stories.
 
         Verifies if the intents used in the stories are valid, and whether
         all valid intents are used in the stories."""
 
-        everything_is_alright = self.verify_intents()
+        everything_is_alright = self.verify_intents(ignore_warnings)
+        everything_is_alright = (
+            self.verify_example_repetition_in_intents(ignore_warnings)
```

Comment:
to me this is checking repetition is independent from checking usage of intents in stories. Let's call  `self.verify_example_repetition_in_intents` in `verify_all` method

Reply:
@Ghostvv, I moved the call of `verify_example_repetition_in_intents` to `verify_all` and added an info log to maintain consistency of how methods are called inside `verify_all`.